### PR TITLE
CFE-2557: Mac platform in travis

### PR DESCRIPTION
**Pull request by olehermanse**

### CODE REVIEW

**Comment by olehermanse** [comment]
(Do not merge yet)

File: .travis.yml [hunk 1]
```diff
@@ -20,104 +20,54 @@ env:
...
+#      env: JOB_TYPE=acceptance_tests_common
+#    - os: osx
+#      sudo: required
+#      env: JOB_TYPE=acceptance_tests_unsafe_serial_network_etc
+
+before_install: ./travis-scripts/before_install.sh
+
+script: ./travis-scripts/script.sh
+
+before_deploy: ./travis-scripts/before_deply.sh
```
Comment (jimis) [inline .travis.yml]: typo?

File: travis-scripts/before_deploy.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+set -ev
+echo "Starting travis bash script: before_deploy.sh"
+# A tag ending with "number.number" is not a prerelease
+if echo $DIST_TARBALL | grep -E -q '[0-9]{1,2}\.[0-9]{1,2}\.tar\.gz';
+then
+    IS_PRERELEASE=false;
+else
+    IS_PRERELEASE=true;
+fi;
+export IS_PRERELEASE
```
Comment (jimis) [inline travis-scripts/before_deploy.sh]: Exporting a variable means that the *children* of this process get it, not the parent. So the parent travis script will not see IS_PRERELEASE. We need to figure out a different way for this.

**Comment by olehermanse** [comment]
I don't know why the mac build still hangs after make check. I think I will try to fix the failing unit tests next.

**Comment by jimis** [comment]
@olehermanse what's the state of this? Was it hanging for unknown reason on the macs but everything else was good?

**Comment by olehermanse** [comment]
> @olehermanse what's the state of this? Was it hanging for unknown reason on the macs but everything else was good?

@jimis Yes, IIRC, all enabled unit tests pass on mac, but it hangs after finishing them. Can pick this up after #2791 and #2801

**Comment by olehermanse** [comment]
@nickanderson Check this out

File: travis-scripts/before_deploy.sh (hunk 2 above)
Comment (olehermanse) [inline travis-scripts/before_deploy.sh]: script is sourced, and have confirmed that exported variables are accessible from other parts of .travis.yml

**Comment by olehermanse** [comment]
Replaced this PR by: https://github.com/cfengine/core/pull/2808

**Pull request closed without merging**